backend/src/gpml/handler/detail.clj

```diff
@@ -268,12 +269,20 @@
 
 (defmethod ig/init-key ::get [_ {:keys [db]}]
   (cache-hierarchies! (:spec db))
-  (fn [{:keys [path-params]}]
-    (if-let [data (db.detail/get-detail (:spec db) (update path-params :topic-id #(Long/parseLong %)))] ;; TODO: investigate why id value is not coerced
+  (fn [{{:keys [path]} :parameters jwt-claims :jwt-claims}]
+    (let [conn (:spec db)
+          topic (:topic-type path)
+          public-topic? (not (contains? constants/approved-user-topics topic))
+          authorized? (or public-topic?
+                          (and (:email jwt-claims)
+                               (= "APPROVED"
```

Comment:
we have already some logic to know if we have an approved user in the `gpml.auth` namespace.
https://github.com/akvo/unep-gpml/blob/6b49d1cd0da768d7c4b3f5d543a615d9b57444b8/backend/src/gpml/auth.clj#L96-L98

I understand that the other middleware *forces* an authenticated user, and here we're dealing with different case? 🤔 

Can we somehow move the `APPROVED` logic to `auth.gpml` namespace?